### DISCUSSION TITLE
Why does `gramineproject/gramine:latest` Docker image use Ubuntu 20.04 instead of 22.04?

### DISCUSSION THREAD

**Discussion by pbeza** [gramineproject/gramine#1710] [General]
Is there any good reason why [`gramineproject/gramine:latest`][gramine-latest] Docker image is still based on _Ubuntu 20.04_ instead of _Ubuntu 22.04_?

I was a bit surprised to find out that I had to use [`gramineproject/gramine:1.6-jammy`][gramine-1.6-jammy] (or [`gramineproject/gramine:stable-jammy`][gramine-stable-jammy] which is the same thing) instead of `gramineproject/gramine:latest`.

[gramine-latest]: https://hub.docker.com/layers/gramineproject/gramine/latest/images/sha256-045e6ecbae0cbb4e41a777a0ecf9100cbfbe7d2a0c8d7c1766a592a48fc82056?context=explore
[gramine-1.6-jammy]: https://hub.docker.com/layers/gramineproject/gramine/1.6-jammy/images/sha256-8ed52f367dcc00d0cf417d427ab48507c0bf34e17805253bbeed3c76ed64b3d1?context=explore
[gramine-stable-jammy]: https://hub.docker.com/layers/gramineproject/gramine/stable-jammy/images/sha256-8ed52f367dcc00d0cf417d427ab48507c0bf34e17805253bbeed3c76ed64b3d1?context=explore

**Comment by dimakuv** [comment]
IIRC, we decided on this purely for backwards compatibility. Otherwise e.g. CI automated jobs could break for people.

@oshogbo Do I remember correctly?

**Reply by mkow** [reply]
Hmm, but we will need to update it someday anyways, so such breakages are unavoidable, so why do it later?

**Comment by dimakuv** [comment]
Ok, I found our private discussion on this:

> &ndash; I would prefer if it was announced in release notes as one of incompatible changes that we change underlying image to 22.04 and then publish 1.6 on 22.04 as latest, but too late for that I guess
> &ndash; Yes. I guess I agree we should announce that. So lets agree that for 1.7 we change the release.

TLDR: We should have announced it first in our Release Notes. So we'll change the `latest` to 22.04 in the next release, together with the announcement.

**Comment by oshogbo** [comment]
Yes, generally, 'latest' refers to the most recent Gramine version, not necessarily the underlying OS. However, we have agreed on upgrading to a newer Ubuntu version. We hadn't realized this before the release, and afterwords we were concerned that making such a change post-release could potentially cause issues with user's Dockerfiles.

In the end, we agreed that we should make a proper announcement and then switch to the new Ubuntu version.

**Comment by pbeza** [comment]
Thanks for the clarification! 🙏🏼
